### ISSUE TITLE
Wrong payment cancelled when cancelling recurring payments for a contact where a contact has two or more ongoing contributions

### ISSUE THREAD

**Issue by kasiawaka** [iATSPayments/com.iatspayments.civicrm#116]
Working on CiviCRM 4.6 with IATS extension 1.4.2
We were running some tests and accidentally discovered that if a contact had two recurring contributions running at the same time, the cancel links that are sent in receipts after each payment (we've enabled that option in the extension configuration) sometimes cancel the wrong contribution. Please let me know if you need more information or tests performed somewhere to verify this issue.

**Comment by adixon** [comment]
Yes, I've heard of a version of this happening - when you try to cancel one of the two recurring contributions, the other one gets cancelled. I'm fairly certain this is a core bug.

Another funny thing is that recurring contributions with no associated contributions can not be edited (you get some kind of sql error), and I suspect these two are related.
